applications/GeoMechanicsApplication/tests/cpp_tests/test_activate_model_part_operation.cpp

```diff
@@ -0,0 +1,76 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+// Project includes
+#include "containers/model.h"
+#include "geometries/quadrilateral_2d_4.h"
+#include "includes/expect.h"
+#include "processes/structured_mesh_generator_process.h"
+#include "testing/testing.h"
+
+// Application includes
+#include "custom_operations/activate_model_part_operation.h"
+
+namespace Kratos::Testing {
+
+KRATOS_TEST_CASE_IN_SUITE(ActivateModelPartOperation, KratosGeoMechanicsFastSuite)
+{
+    // Create the test model part
+    Model test_model;
+    auto& r_test_model_part = test_model.CreateModelPart("TestModelPart");
+
+    // Set up the test model part mesh
+    auto p_point_1 = Kratos::make_intrusive<Node>(1, 0.0, 0.0, 0.0);
+    auto p_point_2 = Kratos::make_intrusive<Node>(2, 0.0, 1.0, 0.0);
+    auto p_point_3 = Kratos::make_intrusive<Node>(3, 1.0, 1.0, 0.0);
+    auto p_point_4 = Kratos::make_intrusive<Node>(4, 1.0, 0.0, 0.0);
+    Quadrilateral2D4<Node> domain_geometry(p_point_1, p_point_2, p_point_3, p_point_4);
+
+    Parameters mesher_parameters(R"({
+        "number_of_divisions": 2,
+        "element_name": "Element2D3N",
+        "condition_name": "LineCondition2D2N",
+        "create_skin_sub_model_part": false
+    })");
+    StructuredMeshGeneratorProcess(domain_geometry, r_test_model_part, mesher_parameters).Execute();
```

Comment:
Its interesting to see how you use this process, thanks for showing this! I have one question about it: I don't see that the process adds conditions to the model part, so it might be that this test loops over the conditions, but there might not be any (so then it also doesn't test them). Could we add a check to the test (same for the de-activate) for the number of nodes/elements/conditions (i.e. KRATOS_EXPECT_TRUE(r_test_model_part.Conditions().size() > 0))? Then we're sure we're testing everything.

Reply:
Yes, the `StructuredMeshGeneratorProcess` is quite handy for building unittests of this sort without the need of having an mdpa. About the test modification, sure.

Reply:
Actually I switched on the creation of the skin (see the `"create_skin_sub_model_part": true`) in order to test the conditions activation/deactivation as well.

Reply:
Thanks, then I think this test is okay!